File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/engine/ConfigRepository.kt
package com.scribblefit.feature.ai.domain.engine

import com.scribblefit.feature.ai.domain.model.SystemConfig
import kotlinx.coroutines.flow.Flow

interface ConfigRepository {
    fun getConfig(): Flow<SystemConfig?>
    suspend fun updateConfig(config: SystemConfig)
    suspend fun syncMetadata(): Result<Unit>
}

File: apps/ios/ScribbleFit/LocalPackages/Core/CoreModel/Sources/CoreModel/ConfigRepository.swift
import Foundation
import Combine

@MainActor
public protocol ConfigRepository: Sendable {
    func getConfig() -> SystemConfig
    func updateConfig(_ config: SystemConfig)
    func resetConfig()
    func syncMetadata() async throws
    var configPublisher: AnyPublisher<SystemConfig, Never> { get }
}
